src/openforms/submissions/tasks/cleanup.py

```diff
@@ -46,12 +46,18 @@ def cleanup_on_completion_results():
 
     cutoff = timezone.now() - timedelta(days=RETAIN_RESULTS_NUM_DAYS)
 
-    submissions = Submission.objects.filter(
-        # only consider submissions that are completed and old enough
-        completed_on__lte=cutoff,
-        suspended_on__isnull=True,
-        # only clean up submissions that have task_ids stored (prevent re-processing)
-        on_completion_task_ids__len__gt=0,
+    submissions = (
+        Submission.objects.annotate(
+            number_tasks_ids=F("postcompletiontasksids__tasks_ids__len")
+        )
+        .filter(
+            # only consider submissions that are completed and old enough
+            completed_on__lte=cutoff,
+            suspended_on__isnull=True,
+            # only clean up submissions that have task_ids stored (prevent re-processing)
+            number_tasks_ids__gt=0,
```

Comment:
I think this related query could be done directly inside the filter without annotate, but I'm not sure.

```suggestion
            # only clean up submissions that have task_ids stored (prevent re-processing)
            postcompletiontasksids__tasks_ids__len__gt=0,
```